tls/s2n_client_hello.c

```diff
@@ -342,15 +344,22 @@ int s2n_client_hello_recv(struct s2n_connection *conn)
     }
     GUARD(s2n_process_client_hello(conn));
 
-    /* s2n_conn_set_handshake_type() is called by SERVER_SESSION_LOOKUP in < TLS 1.3,
-     * which is something not present in the current s2n tls 1.3 state machine.
-     * we call this manually so the state machine can transition to the
-     * negotiated and handshake type for tls1.3
-     */
-    if (conn->actual_protocol_version == S2N_TLS13) {
-        GUARD(s2n_conn_set_handshake_type(conn));
+    return 0;
+}
+
+int s2n_client_hello_status_recv(struct s2n_connection *conn)
```

Comment:
Mentioned offline -- this does not belong in s2n_client_hello, and should be moved to its own module.

Reply:
I moved this into a module called s2n_establish_session. This is because there is no clear location to update the handshake based on CLIENT_HELLO parameters. This change adds the new s2n_establish_session() function to handle a CLIENT_HELLO message. After handling the message, it generates new security parameters to update the handshake OR retrieves those parameters from the cache based on session ID.